### PR TITLE
[hmac,doc] Update documentation

### PR DESCRIPTION
**Pull request by gdessouky**
This documents the new HMAC wider digest and configurable key length feature and updates the block diagrams minimally. This would close issue #21711.

### CODE REVIEW

**Review by andreaskurth — APPROVED** [review]
LGTM; thx for these updates @gdessouky !

**Comment by andreaskurth** [comment]
This PR changes documentation only, so the failing _CW310 Test ROM Tests_ CI check can be ignored. Merging

File: hw/ip/hmac/doc/programmers_guide.md [hunk 1]
```diff
@@ -72,16 +68,17 @@ void run_hmac(uint32_t *msg, uint32_t msg_len, uint32_t *hash) {
 
 ## Updating the configurations
 
-The HMAC IP prevents [`CFG`](registers.md#cfg) and [`KEY`](registers.md#key) registers from updating while the engine is processing messages.
+The HMAC IP prevents [`CFG`](registers.md#cfg) and [`KEY`](registers.md#key) registers from getting updating while the engine is processing messages.
```
Comment (cdgori) [inline hw/ip/hmac/doc/programmers_guide.md:71]: nit: updating -> updated

**Review by cdgori — COMMENTED** [review]
LGTM modulo one doc nit

File: hw/ip/hmac/doc/programmers_guide.md (hunk 1 above)
Comment (gdessouky) [reply hw/ip/hmac/doc/programmers_guide.md:71]: thx, will fix in upcoming PR!

**Pull request merged**